ase/sw/ase_ops.c

```diff
@@ -758,6 +758,16 @@ void ase_memcpy(void *dest, const void *src, size_t n)
 	ase_memcpy_s(dest, n, src, n);
 }
 
+/*
+* ase_memset - Secure memset abstraction
+*/
+void ase_memset(void *dest, int ch, size_t n)
+{
+	// Secure implementation
+	ase_memset_s(dest, n, ch, n);
```

Comment:
I assume you realize that, just like all these other "ase" functions, the majority of the protection offered is eliminated by passing _n_ to both the _dmax_ and _count_ fields. We are satisfying Klocwork but not accomplishing much.

Reply:
Agree.  We might need to switch to using SDL safe string in the future.

Reply:
That won't help. It has the same problem. The problem is we are asserting the length of the target buffer matches the length of the source, which is exactly what is supposed to be tested. Unless we have the targets being smart objects that know their length (which likely means writing in C++ instead of C), this exercise is pointless.